### DISCUSSION TITLE
[Camera Support]:

### DISCUSSION THREAD

**Discussion by bdaniel7** [blakeblackshear/frigate#12259] [Camera Support]
### Describe the problem you are having

ffmpeg input path "tcp/h264://10.0.0.37:8494" is not read by ffmpeg.

When I run "rpicam-vid --level 4.2 -t 0 --width 1280 --height 720 --framerate
      60 --denoise cdn_off --nopreview --codec h264 --profile high -n --listen -o
      tcp://10.0.0.37:8494" in terminal, the video stream starts, then I can connect using vlc, network stream tcp/h264://10.0.0.37:8494

### Version

0.13.2-6476f8a

### Frigate config file

```yaml
mqtt:
  enabled: false

ffmpeg:
  hwaccel_args: preset-rpi-64-h264

go2rtc:
  streams:
    logitech: ffmpeg:device?video=/dev/video0&input_format=h264&video_size=1280x720
    raspicam: exec:rpicam-vid --level 4.2 -t 0 --width 1280 --height 720 --framerate
      60 --denoise cdn_off --nopreview --codec h264 --profile high -n --listen -o
      tcp://10.0.0.37:8494    ## this does not work in Frigate container.

cameras:  
  logitech: ## Logitech HD Pro Webcam C920
    enabled: true
    ffmpeg:
      inputs:
      - path: rtsp://10.0.0.37:8554/logitech  ## this works
        roles:
        - detect
        - record
        - audio
    detect:
      enabled: true
      width: 1280
      height: 720
    zones:
      zone_0:
        coordinates: 921,446,1006,504,1003,663,815,673,737,605,730,456

    motion:
      mask:
      - 511,109,686,107,689,333,689,558,508,559,508,343
  raspicam: ## 
    enabled: true
    ffmpeg:
      inputs:
      - path: tcp/h264://10.0.0.37:8494  ## **this does not work**
        roles:
        - detect
        - record
    detect:
      enabled: true
      width: 1280
      height: 720
```

### Relevant log output

```shell
2024-07-02 11:07:15.632660819  [2024-07-02 11:07:15] watchdog.raspicam              ERROR   : Ffmpeg process crashed unexpectedly for raspicam.
2024-07-02 11:07:15.633171426  [2024-07-02 11:07:15] watchdog.raspicam              ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-07-02 11:07:15.633545553  [2024-07-02 11:07:15] ffmpeg.raspicam.detect         ERROR   : tcp/h264://10.0.0.37:8494: No such file or directory
2024-07-02 11:07:15.933041934  [2024-07-02 11:07:15] frigate.video                  ERROR   : raspicam: Unable to read frames from ffmpeg process.
2024-07-02 11:07:15.933760336  [2024-07-02 11:07:15] frigate.video                  ERROR   : raspicam: ffmpeg process is not running. exiting capture thread...
```

### FFprobe output from your camera

```shell
> ffprobe rtsp://10.0.0.37:8554/logitech
> ffprobe version 5.1.5-0+rpt1+deb12u1 Copyright (c) 2007-2024 the FFmpeg developers
  built with gcc 12 (Debian 12.2.0-14)

[rtsp @ 0x55aa40eeb0] method SETUP failed: 461 Unsupported transport
Input #0, rtsp, from 'rtsp://10.0.0.37:8554/logitech':
  Metadata:
    title           : go2rtc/1.9.2
  Duration: N/A, start: 0.040078, bitrate: N/A
  Stream #0:0: Video: h264 (Constrained Baseline), yuvj420p(pc, progressive), 1280x720 [SAR 1:1 DAR 16:9], 30 fps, 24 tbr, 90k tbn


> ffprobe tcp://10.0.0.37:8494
tcp/h264://10.0.0.37:8494: No such file or directory
```

### Operating system

Debian

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Logitech HD Pro Webcam C920  (USB) + Camera Module 2 Noir

### Any other information that may be helpful

While preparing this report, I kept debugging, using the go2rtc api endpoint (:1984), and I got this error for the raspicam stream:
streams: exec: "rpicam-vid": executable file not found in $PATH.

So then I remembered that Frigate runs in a container, and rpicam/libcamera tools are likely not installed.
So I installed libcamera tools, then ran 
$ docker exec -it 10828b9e6f98 libcamera-vid
Preview window unavailable
[1:03:09.356384450] [5162]  INFO Camera camera_manager.cpp:297 libcamera v0.0.5+83-bde9b04f
ERROR: *** no cameras available ***

$ docker exec -it 10828b9e6f98 dmesg | grep -i video
[    8.675441] bcm2835_vc_sm_cma_probe: Videocore shared memory driver
[    8.931503] videodev: Linux video capture interface: v2.00
[    9.140455] rpivid feb10000.codec: Device registered as /dev/video19
[    9.143104] bcm2835-codec bcm2835-codec: Device registered as /dev/video10
[    9.149645] bcm2835-codec bcm2835-codec: Device registered as /dev/video11
[    9.183368] bcm2835-isp bcm2835-isp: Device node output[0] registered as /dev/video13
[    9.188882] bcm2835-codec bcm2835-codec: Device registered as /dev/video12
[    9.189054] bcm2835-isp bcm2835-isp: Device node capture[0] registered as /dev/video14
[    9.190031] bcm2835-isp bcm2835-isp: Device node capture[1] registered as /dev/video15
[    9.194868] bcm2835-isp bcm2835-isp: Device node stats[2] registered as /dev/video16
[    9.195972] bcm2835-codec bcm2835-codec: Device registered as /dev/video18
[    9.202141] bcm2835-isp bcm2835-isp: Device node output[0] registered as /dev/video20
[    9.204797] bcm2835-isp bcm2835-isp: Device node capture[0] registered as /dev/video21
[    9.205196] bcm2835-codec bcm2835-codec: Device registered as /dev/video31
[    9.223959] bcm2835-isp bcm2835-isp: Device node capture[1] registered as /dev/video22
[    9.231100] bcm2835-isp bcm2835-isp: Device node stats[2] registered as /dev/video23
[    9.485993] usbcore: registered new interface driver uvcvideo

**Comment by NickM-27** [comment]
should be using 127.0.0.1:8554 and not the host IP

**Reply by bdaniel7** [reply]
The trouble is with raspicam stream and camera.
